docs/output.md

```diff
@@ -258,6 +258,91 @@ print(result.output)
 
 _(This example is complete, it can be run "as is")_
 
+#### Handling partial output
+
+When streaming with `run_stream()`, output validators and output functions are called **multiple times** — once for each partial output received from the model, and once for the final complete output.
```

Comment:
We'll need to have something in the output validators section as well. I think it will make sense to split this up into a separate subsection for each feature (with some duplicated text about `run_stream` etc)